### ISSUE TITLE
unexpected property 'prune-backups' (500) When creating Storage

### ISSUE THREAD

**Issue by chanster** [TheGrandWazoo/freenas-proxmox#85]
- Proxmox Version: `6.3-2`
- TrueNAS: `TrueNAS-12.0-RELEASE`
- FreeNAS-Proxmox: `stable` APT repo

### Steps to reproduce
1. From web Proxmox Web Console. click on **datacenter > Storage**
1. Click **Add > ZFS over iSCSI**
1. Fill in details, then click **Add**
1. Error Pops up `unexpected property 'prune-backups' (500) When creating Storage`
1. Click on **Backup Retention**
1. Uncheck **Keep add Backups**
1. Click Add.

Storage is now added. Though it fails to find iSCSI target when cloning or creating new VM.

**Comment by chanster** [comment]
I am unable to edit this from the web console as any change will not save, I get the same error. I had to go into the `storage.conf` file on the server itself to make changes.

**Comment by TheGrandWazoo** [comment]
I have confirmed this on adding a new FreeNAS storage using FreeNAS 11.3 or 12.0. Looks like an underlying property was added to the Plugin.pm and other storage types but not to ANY ZFS or ZFS over iSCSI. Give me a bit and I will see if there is a quick fix for this or a bug in Proxmox VE.

**Comment by TheGrandWazoo** [comment]
I have confirmed that this is an underlying issue with Proxmox VE 6.3-2. A fresh install of Proxmox VE WITHOUT the installation of the FreeNAS plugin causes the same issue 
```
unexpected property 'prune-backups' (500)
```
with Comstar, istgt, IET and LIO.

**Comment by TheGrandWazoo** [comment]
Ok. Looking at the new "Backup Retention" tab on the "ZFS over iSCSI" modal I noticed that it is enabled where the if you look at the "ZFS" modal you will get "Backup content type not available for this storage." so we need that on the "ZFS over iSCSI". 
If you don't mind editing a file you can quickly fix this because the 'ZFS' storage does not support backups from discussions.

This is NOT a plugin issue but an underlying issue with the 'ZFS over iSCSI' script...this will also fail with a virgin install of Proxmox VE using any of the other Comstar, istgt, IET and LIO.


First make a backup copy of the following file incase you need to revert.
```
/usr/share/pve-manager/js/pvemanagerlib.js
```
Then edit the above file with whatever editor you wish.
Then search for the following code (should be around line 46667 if you have patched for the FreeNAS plugin)
```
            {
                xtype: 'textfield',
                name: 'freenas_user',
                reference: 'freenas_user_field',
                inputId: 'freenas_user_field',
                value: '',
                fieldLabel: gettext('API Username'),
                bind: {
                    hidden: '{!isFreeNAS}'
                }
            },
        ];

        me.column2 = [
            {
                xtype: me.isCreate ? 'pveiScsiProviderSelector' : 'displayfield',
                name: 'iscsiprovider',
                value: 'comstar',
                fieldLabel: gettext('iSCSI Provider'),
                allowBlank: false
            },

```
Add the following code
```
            {
                xtype: 'pveContentTypeSelector',
                cts: ['images'],
                fieldLabel: gettext('Content'),
                name: 'content',
                value: ['images'],
                multiSelect: false,
                allowBlank: false
            },
```
So the that section looks like this.
```
            {
                xtype: 'textfield',
                name: 'freenas_user',
                reference: 'freenas_user_field',
                inputId: 'freenas_user_field',
                value: '',
                fieldLabel: gettext('API Username'),
                bind: {
                    hidden: '{!isFreeNAS}'
                }
            },
            {
                xtype: 'pveContentTypeSelector',
                cts: ['images'],
                fieldLabel: gettext('Content'),
                name: 'content',
                value: ['images'],
                multiSelect: false,
                allowBlank: false
            },
        ];

        me.column2 = [
            {
                xtype: me.isCreate ? 'pveiScsiProviderSelector' : 'displayfield',
                name: 'iscsiprovider',
                value: 'comstar',
                fieldLabel: gettext('iSCSI Provider'),
                allowBlank: false
            },
```

Save the changes and then reload the PVE Daemon and Proxy
```
pvedaemon restart
pveproxy restart
```

This will allow you to add/edit you FreeNAS ZFS over iSCSI provide or any other ZFS over iSCSI provider without getting the 
```
unexpected property 'prune-backups' (500)
```
error.

You will also notice with selection the "Backup Retention" tab that the underlying modal is disabled with a message of
```
Backup content type not available for this storage.
```
Hope this helps.

**Comment by chanster** [comment]
Thanks! this worked. I submitted the issue to Proxmox.

**Comment by TheGrandWazoo** [comment]
FYI. It looks like this is fixed in Proxmox VE 6.3-3 under the pve-manager 6.3-2 to 6.3-3 with a backup masking fix.

**Comment by michaeldexter** [comment]
I set up a Proxmox VE 6.3 evaluation system today and the online update does not appear to provide 6.3-3. Is this only available with a license?
